tests/integration-tests/conftest.py

```diff
@@ -893,6 +896,41 @@ def create_role(trusted_service, policies=()):
         iam_client.delete_role(RoleName=role_name)
 
 
+@pytest.fixture(scope="class")
+def instance_profile_factory(region, role_factory):
+    instance_profiles = []
+    iam_client = boto3.client(
+        "iam",
+        region_name=region,
+        config=Config(
+            retries={
+                "max_attempts": 10,
+            }
+        ),
+    )
+
+    def create_instance_profile(policies=()):
+        instance_profile_name = f"integ-tests_{region}_{random_alphanumeric()}"
+        logging.info(f"Creating instance profile {instance_profile_name}")
+        instance_profile_arn = iam_client.create_instance_profile(InstanceProfileName=instance_profile_name)[
+            "InstanceProfile"
+        ]["Arn"]
+        role_name, _ = role_factory("ec2", policies)
+        iam_client.add_role_to_instance_profile(InstanceProfileName=instance_profile_name, RoleName=role_name)
+        logging.info(f"Iam profile is ready: {instance_profile_arn}")
+        instance_profiles.append({"profile_name": instance_profile_name, "role_name": role_name})
+        return instance_profile_arn
+
+    yield create_instance_profile
+
+    for instance_profile in instance_profiles:
+        profile_name = instance_profile["profile_name"]
```

Comment:
The code to delete instance profile must be into the loop, not outside. 
Now you're just deleting the latest profile_name.

Reply:
Good catch! Fixed